tests/flag_util_test.py

```diff
@@ -100,3 +100,84 @@ def testSerialize(self):
 
     self.assertEqual(ser.Serialize(il),
                      '1,2-5,9')
+
+
+class TestObjectSize(unittest.TestCase):
+  def testStrSmall(self):
+    size = flag_util.ObjectSize(bytes=3)
+
+    self.assertEquals(str(size), '3B')
+
+  def testStrBase10(self):
+    size = flag_util.ObjectSize(bytes=3 * (1000 ** 2))
+
+    self.assertEquals(str(size), '3MB')
+
+  def testStrBase2(self):
+    size = flag_util.ObjectSize(bytes=3 * (1024 ** 2))
+
+    self.assertEquals(str(size), '3MiB')
+
+  def testEq(self):
+    self.assertEquals(
+        flag_util.ObjectSize(bytes=3),
+        flag_util.ObjectSize(bytes=3))
+
+  def testLessThan(self):
+    self.assertLess(
+        flag_util.ObjectSize(bytes=3), flag_util.ObjectSize(bytes=5))
+
+  def testGreaterThan(self):
+    self.assertGreater(
+        flag_util.ObjectSize(bytes=5), flag_util.ObjectSize(bytes=3))
+
+  def testExactKiB(self):
+    size = flag_util.ObjectSize(bytes=3 * 1024)
+    self.assertEquals(size.ExactKiB(), 3)
+
+  def testExactKiBInvalid(self):
+    size = flag_util.ObjectSize(bytes=42)
+    with self.assertRaises(ValueError):
+      size.ExactKiB()
+
+  def testAsKib(self):
+    size = flag_util.ObjectSize(bytes=1024 + 512)
+    self.assertEquals(size.AsKiB(), 1.5)
+
+
+class TestParseObjectSize(unittest.TestCase):
+  def testNoInteger(self):
+    with self.assertRaises(ValueError):
+      flag_util.ParseObjectSize('foo')
+
+  def testNoSuffix(self):
+    with self.assertRaises(ValueError):
+      flag_util.ParseObjectSize('12')
+
+  def testJustBytes(self):
+    self.assertEquals(flag_util.ParseObjectSize('12B'),
+                      flag_util.ObjectSize(bytes=12))
+
+  def testGoodSuffix(self):
+    self.assertEquals(flag_util.ParseObjectSize('12MB'),
+                      flag_util.ObjectSize(bytes=12 * (1000 ** 2)))
+
+  def testBadSuffix(self):
+    with self.assertRaises(ValueError):
+      flag_util.ParseObjectSize('12j')
+
+  def testBase2Suffix(self):
+    self.assertEquals(flag_util.ParseObjectSize('12MiB'),
+                      flag_util.ObjectSize(bytes=12 * (1024 ** 2)))
+
+  def testBadSuffixModifier(self):
+    with self.assertRaises(ValueError):
+      flag_util.ParseObjectSize('12Mj')
+
+  def testExtraCharacters(self):
+    with self.assertRaises(ValueError):
+      flag_util.ParseObjectSize('12mikdj')
```

Comment:
This isn't actually extra characters? The first few characters are still not a valid pattern.